### PR TITLE
Reproducible risc0 guest code builds

### PR DESCRIPTION
**Pull request by kpp**
# Description
Reproducible risc0 guest code builds

## Linked Issues
- Fixes https://github.com/chainwayxyz/citrea/issues/637

## Docs

Add env var `GUEST_BUILD_NO_DOCKER` to build risc0 guest code natively

### CODE REVIEW

**Review by eyusufatik — APPROVED** [review]
please also update docs with GUEST_BILD_NO_DOCKER stuff

File: bin/citrea/Cargo.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ ethereum-rpc = { path = "../../crates/ethereum-rpc" }
 sequencer-client = { path = "../../crates/sequencer-client" }
 
 # Sovereign-SDK deps
+citrea-risc0 = { package = "risc0", path = "./provers/risc0" }
```
Comment (rakanalh) [inline bin/citrea/Cargo.toml:26]: This one and `citrea-risc0-bonsai-adapter` should go up into `#Citrea Deps`

File: bin/citrea/provers/risc0/guest-bitcoin/src/bin/bitcoin_da.rs [hunk 2]
```diff
@@ -24,9 +24,9 @@ fn report_bench_metrics(start_cycles: usize, end_cycles: usize) {
     serialized.extend(&size_bytes);
 
     // calculate the syscall name.
-    let cycle_string = String::from("cycle_metrics\0");
+    let name = c"cycle_metrics".as_ptr();
     let metrics_syscall_name =
-        risc0_zkvm_platform::syscall::SyscallName::from_bytes_with_nul(cycle_string.as_ptr());
+        unsafe { risc0_zkvm_platform::syscall::SyscallName::from_bytes_with_nul(name as _) };
```
Comment (rakanalh) [inline bin/citrea/provers/risc0/guest-bitcoin/src/bin/bitcoin_da.rs:29]: I am curious... why did you change this to unsafe though the previous version did not use the unsafe version?

Comment (kpp) [reply bin/citrea/provers/risc0/guest-bitcoin/src/bin/bitcoin_da.rs:29]: Sure, I can fix that. In 1.0.0-rc they didn't have a safe API, that's the code from that time.

**Review by rakanalh — APPROVED** [review]

**Review by eyusufatik — APPROVED** [review]
